File: apis/src/main/java/com/beat/apis/schedule/api/ScheduleController.java
package com.beat.apis.schedule.api;

import org.springframework.http.HttpStatus;
import org.springframework.http.ResponseEntity;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.PathVariable;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RequestParam;
import org.springframework.web.bind.annotation.RestController;

import com.beat.apis.schedule.application.dto.response.TicketAvailabilityResponse;
import com.beat.apis.schedule.api.response.ScheduleSuccessCode;
import com.beat.apis.schedule.facade.ScheduleFacade;
import com.beat.global.support.response.SuccessResponse;

import lombok.RequiredArgsConstructor;

@RestController
@RequestMapping("/api/schedules")
@RequiredArgsConstructor
public class ScheduleController implements ScheduleApi {

	private final ScheduleFacade scheduleFacade;

	@Override
	@GetMapping("/{scheduleId}/availability")
	public ResponseEntity<SuccessResponse<TicketAvailabilityResponse>> getTicketAvailability(
		@PathVariable Long scheduleId,
		@RequestParam int purchaseTicketCount) {

		TicketAvailabilityResponse response = scheduleFacade.findTicketAvailability(scheduleId, purchaseTicketCount);

		return ResponseEntity.status(HttpStatus.OK)
			.body(SuccessResponse.of(ScheduleSuccessCode.TICKET_AVAILABILITY_RETRIEVAL_SUCCESS, response));
	}
}

File: apis/src/main/kotlin/com/beat/apis/schedule/api/ScheduleController.kt
package com.beat.apis.schedule.api

import com.beat.apis.schedule.api.response.ScheduleSuccessCode
import com.beat.apis.schedule.api.response.TicketAvailabilityResponse
import com.beat.apis.schedule.facade.ScheduleFacade
import com.beat.apis.response.SuccessResponse
import org.springframework.http.HttpStatus
import org.springframework.http.ResponseEntity
import org.springframework.web.bind.annotation.GetMapping
import org.springframework.web.bind.annotation.PathVariable
import org.springframework.web.bind.annotation.RequestMapping
import org.springframework.web.bind.annotation.RequestParam
import org.springframework.web.bind.annotation.RestController

@RestController
@RequestMapping("/api/schedules")
class ScheduleController(
    private val scheduleFacade: ScheduleFacade,
) : ScheduleApi {

    @GetMapping("/{scheduleId}/availability")
    override fun getTicketAvailability(
        @PathVariable scheduleId: Long,
        @RequestParam purchaseTicketCount: Int,
    ): ResponseEntity<SuccessResponse<TicketAvailabilityResponse>> {
        val response = scheduleFacade.findTicketAvailability(scheduleId, purchaseTicketCount)
        return ResponseEntity.status(HttpStatus.OK)
            .body(SuccessResponse.of(ScheduleSuccessCode.TICKET_AVAILABILITY_RETRIEVAL_SUCCESS, response))
    }
}
